---
 configure.ac |   16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

Index: libedit/configure.ac
===================================================================
--- libedit.orig/configure.ac
+++ libedit/configure.ac
@@ -46,12 +46,11 @@ AC_PROG_LN_S
 AC_PROG_AWK
 EL_MANTYPE
 
-AC_CHECK_LIB(ncurses, tgetent,,
-  [AC_CHECK_LIB(curses, tgetent,,
-    [AC_CHECK_LIB(termcap, tgetent,,
-      [AC_CHECK_LIB(tinfo, tgetent,,
+AC_CHECK_LIB(termcap, tgetent,,
+  [AC_CHECK_LIB(ncurses, tgetent,,
+    [AC_CHECK_LIB(curses, tgetent,,
+      [AC_MSG_ERROR([libtermcap, libncurses, or libcurses is required!])]
         [AC_MSG_ERROR([libncurses, libcurses, libtermcap or libtinfo is required!])]
-      )]
     )]
   )]
 )
@@ -81,10 +80,10 @@ AC_HEADER_STDC
 AC_HEADER_SYS_WAIT
 AC_CHECK_HEADERS([fcntl.h limits.h malloc.h stdlib.h string.h sys/ioctl.h sys/param.h unistd.h curses.h ncurses.h sys/cdefs.h termcap.h])
 
-AC_CHECK_HEADER([ncurses.h], [],
-  [AC_CHECK_HEADER([curses.h], [],
-    [AC_CHECK_HEADER([termcap.h], [],
-      [AC_MSG_ERROR([ncurses.h, curses.h, or termcap.h is required!])],
+AC_CHECK_HEADER([termcap.h], [],
+  [AC_CHECK_HEADER([ncurses.h], [],
+    [AC_CHECK_HEADER([curses.h], [],
+      [AC_MSG_ERROR([termcap.h, ncurses.h, or curses.h is required!])],
     [])],
   [])],
 [])
